### PR TITLE
Start working with the side menu - Expect the user to have a current eve...

### PR DESCRIPTION
**Pull request by dwhenry**
...nt stored in a session.. otherwise default to the current last created event..

### CODE REVIEW

File: app/site/app/assets/stylesheets/site/application.scss [hunk 1]
```diff
@@ -2,3 +2,7 @@
 @import 'modules/typo';
 @import 'modules/layout';
 @import 'components/data-header';
+
+.menu-item a, .menu-heading a {
```
Comment (plcstevens) [inline app/site/app/assets/stylesheets/site/application.scss]: This shouldn't really be in the main application scss file IMO. What does it belong to?

File: app/site/app/helpers/site/events_helper.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Site
+  module EventsHelper
+    def current_event
```
Comment (plcstevens) [inline app/site/app/helpers/site/events_helper.rb:3]: Why not just put this in application_helper?

**Comment by plcstevens** [comment]
Can't merge, but minor changes ignored, seems okay

File: app/site/app/assets/stylesheets/site/application.scss (hunk 1 above)
Comment (dwhenry) [inline app/site/app/assets/stylesheets/site/application.scss]: just lazy really.. not even 100% sure it is needed...

File: app/site/app/assets/javascripts/site/components/login.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      if(this.state.state === 'loggedOut') {
+        return (<MDE.LoggedOut />);
+      }
+      if(this.state.state === 'login') {
+        return (<MDE.LoginForm  />);
+      }
+    }
+  });
+
+  if(document.getElementById('login')) {
```
Comment (plcstevens) [inline app/site/app/assets/javascripts/site/components/login.js:34]: Is this a common pattern for React? (asking)

File: app/site/app/assets/javascripts/site/components/login/loggedIn.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            <a href='#'>Picks</a>
+          </div>
+          <div className="menu-item">
+            <a href={"/users/" + this.state.userID}>Preferences</a>
+          </div>
+        </div>
+      )
+    },
+    signOut: function() {
+      this.getFlux().actions.logout();
```
Comment (plcstevens) [inline app/site/app/assets/javascripts/site/components/login/loggedIn.js:34]: Quirky

File: app/site/app/assets/javascripts/site/stores/loginStore.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      this.bindActions(
+        MDE.LoginConstants.LOGIN, this.onLogin,
+        MDE.LoginConstants.LOGOUT, this.onLogout,
+        MDE.LoginConstants.CANCEL, this.onCancel,
+        MDE.LoginConstants.SIGN_IN, this.onSignIn
+      );
+    },
+
+    _calculateState: function() {
+      return localStorage.getItem('userID') === null ? 'loggedOut' : 'loggedIn';
```
Comment (plcstevens) [inline app/site/app/assets/javascripts/site/stores/loginStore.js:16]: Well...have to start somewhere!

File: app/site/app/assets/javascripts/site/stores/loginStore.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      this.emit("change");
+    },
+
+    onSignIn: function (payload) {
+      var self = this;
+      $.ajax({
+        url: '/login',
+        method: 'POST',
+        data: payload,
+      }).success(function(response) {
```
Comment (plcstevens) [inline app/site/app/assets/javascripts/site/stores/loginStore.js:37]: Not a biggy but you might be interested in this

```
An alternative construct to the success callback option, the .done() method replaces the deprecated jqXHR.success() method. Refer to deferred.done() for implementation details.
```

File: app/site/app/views/layouts/site/application.html.slim [hunk 7]
```diff
@@ -3,7 +3,7 @@ html lang="en"
 head
   meta charset="utf-8"
 
-  = javascript_include_tag "local_cdn/react-0.12.2", "local_cdn/JSXTransformer-0.12.2"
+  = javascript_include_tag "local_cdn/react-0.12.2", "local_cdn/JSXTransformer-0.12.2", "local_cdn/fluxxor"
```
Comment (plcstevens) [inline app/site/app/views/layouts/site/application.html.slim:6]: Why don't we add these to the asset pipeline? (config/assets.rb) or is it not possible inside Engine?

File: app/site/app/assets/javascripts/site/components/login.js (hunk 3 above)
Comment (dwhenry) [inline app/site/app/assets/javascripts/site/components/login.js:34]: don't think so.. but couldn't find/think of anything better.. open to suggestions.. more research time..

File: app/site/app/assets/javascripts/site/components/login/loggedIn.js (hunk 4 above)
Comment (dwhenry) [inline app/site/app/assets/javascripts/site/components/login/loggedIn.js:34]: try to mix in page events (login) with application event (change view) with not yet implemented stuff.. :)

File: app/site/app/views/layouts/site/application.html.slim (hunk 7 above)
Comment (dwhenry) [inline app/site/app/views/layouts/site/application.html.slim:6]: it is...

**Pull request merged**